rule_test.go

```diff
@@ -474,5 +672,8 @@ func ruleEquals(a, b Rule) bool {
 		a.Invert == b.Invert &&
 		a.Tos == b.Tos &&
 		a.IPProto == b.IPProto &&
-		a.Protocol == b.Protocol
+		a.Protocol == b.Protocol &&
+		a.Mark == b.Mark &&
+		(a.Mask == b.Mask || a.Mask != nil && b.Mask != nil && *a.Mask == *b.Mask ||
+			(a.Mark != 0 && (a.Mask == nil && *b.Mask == 0xFFFFFFFF || b.Mask == nil && *a.Mask == 0xFFFFFFFF)))
```

Comment:
hopefully the `ptrEqual` utility function suggested above can make this more readable

Reply:
I have add func ptrEqual